src/cljx/schema/spec/variant.cljx

```diff
@@ -0,0 +1,57 @@
+(ns schema.spec.variant
+  (:require
+   #+clj [schema.macros :as macros]
+   [schema.utils :as utils]
+   [schema.spec.core :as spec])
+  #+cljs (:require-macros [schema.macros :as macros]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Variant Specs
+
+(defn- option-step [o params else]
+  (let [g (:guard o)
+        c (spec/sub-checker o params)
```

Comment:
I'm not sure I follow, the same params as what?  `cache` should always be in params.